Guidelines.md

```diff
@@ -24,55 +51,41 @@ inside the nuspec file. This is not too hard to accomplish of you write your own
 `dotnet.exe pack` to create the nuget package you need to explicitly mark those references as private assets inside
 the new csproj format.
 
-    <ItemGroup>
-      <PackageReference Include="Cake.Core" Version="0.22.0" PrivateAssets="All" />
-      <PackageReference Include="Cake.Common" Version="0.22.0" PrivateAssets="All" />
-    </ItemGroup>
-
-## Assembly References
```

Comment:
IMHO this is still an important (and often forgotten) issue. The question is do we have a best practice for addins targeting .netstandard (as Fody Costura doesn't support it)?

Reply:
I was under the impression IL merging was no longer necessary. Am I wrong about this?

Reply:
I think it's still an issue, even with the embedded NuGet client, as only one version of an assembly can be loaded to an AppDomain. Which means if AddinA requires LibraryX in version 1.0.0, and don't work with 2.0.0, and AddinB requires LibraryX in version 2.0.0, and won't work with 1.0.0, there will be exceptions.

But I think @mholo65 should be able to confirm this.

Reply:
InProcess NuGet will automatically exclude any Cake.Core or Cake.Common dependencies. Additionally @daveaglick did some improvements regarding assembly resolution when there’s already a version of the assembly loaded in the AppDomain/AssemblyLoadContext. So IL merging or similar approaches (such as embedding with Fody.Custura) *should* not be needed. If you find any bugs regarding dependency loading, please report them.

Reply:
But this works only as long as all addins can work with the version of the loaded assembly? In my example above where there's no common version I assume it still won't work? While this might be an edge case, the other issue is that without IL merging the addin no longer has control with which exact version of the dependencies it will be run, which might lead to hard to diagnose issues for the addin author.

Reply:
True, as long as there are no breaking changes in the API of the dependency shared by the two addins, it *should* work. :smile: